lib/class-wp-rest-widget-areas-controller.php

```diff
@@ -0,0 +1,378 @@
+			$data[ $sidebar_id ] = $this->get_sidebar_data( $sidebar_id );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if a given request has access to read a widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_user_cannot_edit',
+				__( 'Sorry, you are not allowed to edit sidebars.', 'gutenberg' )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves a specific widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		return rest_ensure_response( $this->get_sidebar_data( $request['id'] ) );
+	}
+
+	/**
+	 * Checks if a given REST request has access to update a widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has access to update the item, error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_user_cannot_edit',
+				__( 'Sorry, you are not allowed to edit sidebars.', 'gutenberg' )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Updates a single widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$sidebar_id      = $request->get_param( 'id' );
+		$sidebar_content = $request->get_param( 'content' );
+		global $wp_registered_sidebars;
+
+		$sidebars = wp_get_sidebars_widgets();
+		$sidebar  = $sidebars[ $sidebar_id ];
+		$post_id  = wp_insert_post(
+			array(
+				'ID'           => is_numeric( $sidebar ) ? $sidebar : 0,
+				'post_content' => $sidebar_content,
+				'post_type'    => 'wp_area',
+			)
+		);
+		if ( ! is_numeric( $sidebar ) ) {
+			wp_set_sidebars_widgets(
+				array_merge(
+					$sidebars,
+					array(
+						$sidebar_id           => $post_id,
+						'wp_inactive_widgets' => array_merge(
+							$sidebars['wp_inactive_widgets'],
+							$sidebar
+						),
+					)
+				)
+			);
+		}
+
+		return rest_ensure_response( $this->get_sidebar_data( $request['id'] ) );
+	}
+
+	/**
+	 * Verifies if a sidabar id is valid or not.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param string $sidebar_id Indentifier of the sidebar.
+	 * @return boolean True if the $sidebar_id value is valid and false otherwise.
+	 */
+	public function is_valid_sidabar_id( $sidebar_id ) {
```

Comment:
Wondering if we should make a `WP_Widgets_Manager` class which contains all of the widget-managing logic. Then, `WP_REST_Widget_Areas_Controller` can just contain REST endpoint logic.